### PR TITLE
Add Rabbit68 Keyboard w/ default,kaiec keymaps.

### PR DESCRIPTION
**Pull request by kaiec**
New Keyboard, based on an Open Source PCB.

## Description

Added the Rabbit68 keyboard with two keymaps under keyboards/rabbit68.
The keyboard uses a PCB that is published open source at https://github.com/kaiec/rabbit68.
The PCB uses a Teensy 2.0 controller.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rabbit/rabbit68/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// Defines the keycodes used by our macros in process_record_user
+// enum custom_keycodes { QMKBEST = SAFE_RANGE, QMKURL };
+
+
+#define _BASE 0
+#define _FN 1
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BASE] = LAYOUT(
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, \
```
Comment (fauxpark) [inline keyboards/rabbit/rabbit68/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

File: keyboards/rabbit/rabbit68/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    [_FN] = LAYOUT(
+  _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  _______, _______, _______,  \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,  \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,  \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_VOLD, KC_VOLU, KC_MUTE, _______, _______, _______,  \
+  _______, _______, _______, _______,          _______, _______,          _______, _______, _______, _______, _______, _______ \
+  ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```
Comment (fauxpark) [inline keyboards/rabbit/rabbit68/keymaps/default/keymap.c]: These functions can be removed, as they don't do anything.

File: keyboards/rabbit/rabbit68/keymaps/default/readme.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# The default keymap for rabbit68
+
+Based on this layout: http://www.keyboard-layout-editor.com/#/gists/c885816c6b763c657e40474b015e362b
+
+- US QUERTY Keys
```
Comment (fauxpark) [inline keyboards/rabbit/rabbit68/keymaps/default/readme.md]:
```suggestion
- US QWERTY Keys
```

File: keyboards/rabbit/rabbit68/pins.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+R1: B6
```
Comment (fauxpark) [inline keyboards/rabbit/rabbit68/pins.txt]: This is already documented by the keyboard-level config.h.

File: keyboards/rabbit/rabbit68/readme.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+Rabbit 68
+===
+
+![Rabbit 68 Project](https://github.com/kaiec/rabbit68)
+
+A compact 68 keys ortholinear keyboard.
+
+Keyboard Maintainer: [Kai Eckert](https://github.com/kaiec)  
+Hardware Supported: Rabbit68 PCB rev1
```
Comment (fauxpark) [inline keyboards/rabbit/rabbit68/readme.md]:
```suggestion
Hardware Supported: Rabbit68 PCB rev1  
```

The two spaces at the end of these first two lines are necessary for GitHub to render the section properly.

File: keyboards/rabbit/rabbit68/rules.mk [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
+BOOTLOADER = halfkay
+
+
+# If you don't know the bootloader type, then you can specify the
```
Comment (fauxpark) [inline keyboards/rabbit/rabbit68/rules.mk]: Lines 15 to 38 can be removed; QMK defines these as defaults through `MCU` and `BOOTLOADER`.

File: keyboards/rabbit/rabbit68/readme.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+Hardware Supported: Rabbit68 PCB rev1  
+Hardware Availability: Open Source 
+
+Make example for this keyboard (after setting up your build environment):
+
+    make rabbit/rabbit68:default
+
+Install examples:
+
+    make rabbit/rabbit68:default:teensy
```
Comment (drashna) [inline keyboards/rabbit/rabbit68/readme.md]: Could you use the universal `flash` command here, instead?  

```suggestion
    make rabbit/rabbit68:default:flash
```

It's not necessary, but it's nicer.

Comment (kaiec) [reply keyboards/rabbit/rabbit68/readme.md]: Sure, didn't know about it. Thanks both @fauxpark and @drashna for the feedback!

File: keyboards/rabbit/rabbit68/readme.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+Rabbit 68
+===
+
+![Rabbit 68 Project](https://github.com/kaiec/rabbit68)
```
Comment (drashna) [inline keyboards/rabbit/rabbit68/readme.md]: This should actually be an image: 

```suggestion
![Rabbit 68 Project](https://raw.githubusercontent.com/kaiec/rabbit68/master/images/rabbit68-rev1-rendered-bg.jpg)
```

**Comment by drashna** [comment]
Thanks!

**Pull request merged**